front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -113,6 +125,51 @@ const DEFAULT_ASSISTANT_STATE: AssistantBuilderState = {
   description: null,
   instructions: null,
   avatarUrl: null,
+  modelSettings: {
+    modelId: "gpt-4",
+    modelProviderId: "openai",
+    temperature: 0.7,
+  },
+};
+
+const CREATIVITY_LEVELS = [
+  { label: "Deterministic", value: 0 },
+  { label: "Factual", value: 0.2 },
+  { label: "Balanced", value: 0.7 },
+  { label: "Creative", value: 1 },
+];
+
+const getCreativityLevelFromTemperature = (temperature: number) => {
+  const closest = CREATIVITY_LEVELS.reduce((prev, curr) =>
+    Math.abs(curr.value - temperature) < Math.abs(prev.value - temperature)
+      ? curr
+      : prev
+  );
+  return closest;
+};
+
+const MODELS = [
```

Comment:
Can we real quick move this in api/lib/generation.ts ?

Behind `auth` based function so that we can return based on plan status and centralize there?